pkg/console/operator/sync_v400.go

```diff
@@ -358,6 +366,29 @@ func (co *consoleOperator) SyncConfigMap(
 	return cm, cmChanged, "ConsoleConfigBuilder", cmErr
 }
 
+func (co *consoleOperator) SyncManagedClusterConfigMap(ctx context.Context) (bool, string, error) {
+	managedClusterConfigMap, err := co.configMapClient.ConfigMaps(api.OpenShiftConsoleNamespace).Get(ctx, api.ManagedClusterConfigMapName, metav1.GetOptions{})
+	// Not degraded if get fails because the config map isn't there
+	if apierrors.IsNotFound(err) {
+		klog.V(4).Infof("%v config map not found, continuing...", api.ManagedClusterConfigMapName)
+		return false, "", nil
+	}
+
+	// Degraded if get fails for any other reason
+	if err != nil {
+		return false, "FailedGet", err
+	}
+
+	// Degraded if managed cluster config map is present but doesn't have the correct data key
+	_, ok := managedClusterConfigMap.Data[api.ManagedClusterConfigKey]
+	if !ok {
+		return false, "MissingManagedClusterConfig", fmt.Errorf("%v ConfigMap is missing %v data key", api.ManagedClusterConfigMapName, api.ManagedClusterConfigKey)
+	}
+
```

Comment:
Wondering if we wanna also try to Unmarshal the string as part of the validation

Reply:
The more I think about it, the less I think this check is even necessary. We validate that managed cluster configs are non-empty every step of the way, so it's impossible for us to get an empty value when marshaling. Even if we did somehow pass invalid config to the `DefaultManagedClustersConfigMap` function, it would fail marshaling and the ConfigMap would not be created.